### PR TITLE
Added unit test for strv_env_name_is_valid() function listed in env-util.c

### PR DESCRIPTION
**Pull request by taniishkaaa**
Added unit test for strv_env_name_is_valid() function listed in env-util.c (src/basic)

### CODE REVIEW

**Review by bluca — APPROVED** [review]

**Comment by bluca** [comment]
Next time squash the commits and force push

File: src/test/test-env-util.c [hunk 1]
```diff
@@ -452,4 +452,13 @@ TEST(getenv_steal_erase) {
         assert_se(r > 0);
 }
 
+TEST(strv_env_name_is_valid) {
+        const char *valid_env_names[] = {"HOME", "USER", "SHELL", "PATH", NULL};
+        const char *invalid_env_names[] = {"", "PATH", "home", "user", "SHELL", NULL};
+        const char *repeated_env_names[] = {"HOME", "USER", "SHELL", "USER", NULL};
+        assert_se(strv_env_name_is_valid((char **) valid_env_names));
+        assert_se(!strv_env_name_is_valid((char **) invalid_env_names));
+        assert_se(!strv_env_name_is_valid((char **) repeated_env_names));
```
Comment (daandemeyer) [inline src/test/test-env-util.c:461]: This should use `STRV_MAKE()` instead of constructing the strvs manually.

Comment (taniishkaaa) [reply src/test/test-env-util.c:461]: -> #27108

**Comment by taniishkaaa** [comment]
> Next time squash the commits and force push

I will keep that in mind, thank you!

**Comment by taniishkaaa** [comment]
Closing this PR in favor of #27130

**Pull request merged**